src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,972 @@
+            // in order to update the original expression and containing method.
+            methodCallSitesBuilder.Add(document, new List<SyntaxNode>());
+
+            foreach (var refLocation in referencedLocations)
+            {
+                // Does not support cross-language references currently
+                if (refLocation.Document.Project.Language == document.Project.Language)
+                {
+                    var reference = refLocation.Location.FindNode(cancellationToken).GetRequiredParent();
+                    while (reference is not (TObjectCreationExpressionSyntax or TInvocationExpressionSyntax))
+                    {
+                        reference = reference.GetRequiredParent();
+                    }
+
+                    // Only adding items that are of type InvocationExpressionSyntax or TObjectCreationExpressionSyntax
+                    var invocationOrCreation = reference as TObjectCreationExpressionSyntax ?? (SyntaxNode?)(reference as TInvocationExpressionSyntax);
+                    if (invocationOrCreation is null)
+                    {
+                        continue;
+                    }
+
+                    if (!methodCallSitesBuilder.TryGetValue(refLocation.Document, out var list))
+                    {
+                        list = new List<SyntaxNode>();
+                        methodCallSitesBuilder.Add(refLocation.Document, list);
+                    }
+
+                    list.Add(invocationOrCreation);
+                }
+            }
+
+            return methodCallSitesBuilder.ToImmutable();
+        }
+
+        /// <summary>
+        /// If the parameter is optional and the invocation does not specify the parameter, then
+        /// a named argument needs to be introduced.
+        /// </summary>
+        private static SeparatedSyntaxList<SyntaxNode> AddArgumentToArgumentList(
+            SeparatedSyntaxList<SyntaxNode> invocationArguments, SyntaxGenerator generator,
+            SyntaxNode newArgumentExpression, string parameterName, int insertionIndex, bool named)
+        {
+            var argument = named
+                ? generator.Argument(parameterName, RefKind.None, newArgumentExpression)
+                : generator.Argument(newArgumentExpression);
+            return invocationArguments.Insert(insertionIndex, argument.WithAdditionalAnnotations(Simplifier.Annotation));
+        }
+
+        private static bool ShouldArgumentBeNamed(Compilation compilation, ISemanticFactsService semanticFacts,
+            SemanticModel semanticModel, SeparatedSyntaxList<SyntaxNode> invocationArguments, int methodInsertionIndex,
+            CancellationToken cancellationToken)
+        {
+            var invocationInsertIndex = 0;
+            foreach (var invocationArgument in invocationArguments)
+            {
+                var argumentParameter = semanticFacts.FindParameterForArgument(semanticModel, invocationArgument, cancellationToken);
```

Comment:
can FindParameterForArgument return null?

Reply:
looks like 'yes'.  so be resilient here.

Reply:
added null check